internal/core/src/segcore/SegmentGrowingImpl.h

```diff
@@ -380,14 +380,13 @@ class SegmentGrowingImpl : public SegmentGrowing {
 
     bool
     HasIndex(FieldId field_id) const override {
-        auto& field_meta = schema_->operator[](field_id);
-        if ((IsVectorDataType(field_meta.get_data_type()) ||
-             IsGeometryType(field_meta.get_data_type())) &&
-            indexing_record_.SyncDataWithIndex(field_id)) {
-            return true;
+        if (!schema_->is_field_exist(field_id)) {
+            return false;
         }
-
-        return false;
+        const auto& field_meta = schema_->operator[](field_id);
+        return (IsVectorDataType(field_meta.get_data_type()) ||
+                IsGeometryType(field_meta.get_data_type())) &&
+               indexing_record_.SyncDataWithIndex(field_id);
     }
```

Comment:
on query side, updateSchema will only update Collection-Level schema on delegator and querynode
so in a single time-range window, schema_ in the segment will lack newly added field